src/persist/src/runtime.rs

```diff
@@ -670,21 +670,31 @@ mod tests {
 
         let mut registry = MemMultiRegistry::new();
         let client1 = registry.open("1", "multi")?;
-        let client2 = registry.open("2", "multi")?;
+        // let client2 = registry.open("2", "multi")?;
 
         let (c1s1, c1s1_read) = client1.create_or_load::<String, ()>("1");
         let (c1s2, c1s2_read) = client1.create_or_load::<String, ()>("2");
-        let (c2s1, _) = client2.create_or_load::<String, ()>("1");
+        // let (c2s1, _) = client2.create_or_load::<String, ()>("1");
 
         // Cannot construct with no streams.
-        let no_streams: &[StreamWriteHandle<String, ()>] = &[];
-        assert!(MultiWriteHandle::new_from_streams(no_streams.into_iter()).is_err());
+        // TODO: I'd like to panic here, because creating it with no handles seems to be a bug, not
```

Comment:
there do seem to be two classes of errors involved. one are invalid usage and unrecoverable (runtime id mismatch) and the other is retry-able (once we get to that). i kinda feel like the contract of this thing is that it should return a result from construction for the former (which lets dataflow handle it with a panic if it so chooses) and the latter comes out via seal/atomic_write/etc. thoughts?

Reply:
I think that sounds reasonable, yes. I'll update the PR.